sonic_data_client/db_client.go

```diff
@@ -80,6 +80,23 @@ var IntervalTicker = func(interval time.Duration) <-chan time.Time {
 	return time.After(interval)
 }
 
+// Define a mutex to synchronize access to the IntervalTicker variable
+var intervalTickerMutex sync.Mutex
+
+// Define a new function to set the IntervalTicker variable
+func SetIntervalTicker(f func(interval time.Duration) <-chan time.Time) {
+    intervalTickerMutex.Lock()
+    defer intervalTickerMutex.Unlock()
+    IntervalTicker = f
+}
+
+// Define a new function to get the IntervalTicker variable
+func GetIntervalTicker() func(interval time.Duration) <-chan time.Time {
+    intervalTickerMutex.Lock()
+    defer intervalTickerMutex.Unlock()
+    return IntervalTicker
+}
```

Comment:
Production behavior is different now.
Telemetry didn't use lock for intervalTicker(), now we add a lock mechanism to prevent test code from interfering with production code.
Is it possible to keep production behavior?

Reply:
I think we may not be able to mock this, since product will read global IntervalTicker, in test code we need rewrite it. Mock in test code will change the behavior which breaks the test. Any suggestion?